### ISSUE TITLE
Add upstream CI build

### ISSUE THREAD

**Issue by ravwojdyla** [pystatgen/sgkit#444]
Dask has an "upstream" CI build, which is an interesting take on this problem, it essentially runs dask CI with main/master versions of some of the core dependencies. They run this on schedule as well as on master changes. This could be valuable for us to detect breaking changes before releases. See the upstream CI definition [here](https://github.com/dask/dask/blob/master/.github/workflows/ci-upstream.yml).

Edit: but I'm not sure if it leads to "actions", see the state of that [build](https://github.com/dask/dask/actions?query=workflow%3AUpstream).

_Originally posted by @ravwojdyla in https://github.com/pystatgen/sgkit/issues/422#issuecomment-757365235_

**Comment by ravwojdyla** [comment]
Our current dependencies:

```
numpy
xarray
dask[array] == 2.30.0
dask-ml
scipy
typing-extensions
numba
zarr
fsspec
```

out of those, we should probably test upstream of:

```
xarray
dask[array] 
distributed
zarr
```

as our "core", wdyt?

**Comment by tomwhite** [comment]
I agree those core dependencies are the ones to track.

**Comment by ravwojdyla** [comment]
Closed via #445
